Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM php:8.1.12-cli
+
+RUN apt-get update && apt-get install -y -q git rake ruby-ronn zlib1g-dev && apt-get clean
+
+RUN cd /usr/local/bin && curl -sS https://getcomposer.org/installer | php
+RUN cd /usr/local/bin && mv composer.phar composer
+RUN pecl install grpc
+RUN docker-php-ext-enable grpc
+
+COPY . /app
+
+WORKDIR /app
+
+RUN composer install
```

Comment:
This part is obviously important and I can imagine that maybe it is expensive?

Do you know where composer installs files to?  Are they installed somewhere underneath the current working directory?  Or in a centralized cache that can be re-used by multiple PHP apps on the host?

I'm wondering if we can just have a generic composer.json file that we burn into the image, which would install all of our normal deps and maybe laravel too, and use `composer install` on that file in order to bake all of those deps into the generic image.  And then they could be re-used for various different apps that we might want to test with.  I'm not sure if that's possible or not.  Do you know?

Reply:
I don't think baking cache into a image is not possible but we can cache composer packages for the first build and subsequent builds can export data from that cache.
I think we can use [this](https://docs.docker.com/build/building/cache/backends/).
I'd assume we will have build/push actions on GitHub action so maybe we're interested in using [this](https://docs.docker.com/build/building/cache/backends/gha/)?
Also, we can make it so that only new packages are installed in the subsequent builds by copying `composer.json` and `composer.lock` first and then run `composer install`.

Please let me know what you think!

Reply:
Don't worry about the automated build/push for now, we can tackle that a little later.  but thanks for thinking about it.

As for the caching of composer stuff via that first link you sent, it might be good for us to have a zoom to brainstorm on the ideas there.

The idea of copying in the composer files and then doing composer install is definitely along the lines of what I was thinking.